quantum/pointing_device/pointing_device_auto_mouse.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+#include "pointing_device.h"
+
+/* Check settings and set defaults */
+#ifndef POINTING_DEVICE_AUTO_MOUSE_ENABLE
+#    error "POINTING_DEVICE_AUTO_MOUSE_ENABLE not defined! settings may be incorrect!"
+#endif
+
+#ifndef AUTO_MOUSE_DEFAULT_LAYER
+#    define AUTO_MOUSE_DEFAULT_LAYER 1
+#else
+#    if AUTO_MOUSE_DEFAULT_LAYER > 15
+#        error "AUTO_MOUSE_DEFAULT_LAYER set to number greater than 15! handling of mouse layer keys unsupported"
```

Comment:
Just a heads up, action_layer.h has a define based on the layer state set. So best to check against that. (default is 16, but can be 8 or 32, as well).

```suggestion
#    if AUTO_MOUSE_DEFAULT_LAYER => MAX_LAYER
#        error "AUTO_MOUSE_DEFAULT_LAYER set to number greater the number of supported layers! Handling of mouse layer keys unsupported"
```